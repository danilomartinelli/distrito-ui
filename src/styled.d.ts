import 'styled-components';

declare module 'styled-components' {
  export interface DefaultTheme {
    font: {
      weight: {
        regular: string;
        medium: string;
        bold: string;
      };
    };
    colors: {
      brand: {
        n700: string;
        n600: string;
        n500: string;
        n200: string;
        n100: string;
      };
      neutral: {
        white: string;
        n100: string;
        n200: string;
        n300: string;
        n400: string;
        n500: string;
        n800: string;
      };
    };
    name: string;
  }
}
