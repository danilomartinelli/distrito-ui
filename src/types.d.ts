declare module '*.svg' {
  const content: React.ComponentType<any>;
  export default content;
}
