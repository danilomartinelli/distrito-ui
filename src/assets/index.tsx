export * from './DotsLoading';
export * from './SearchLoupe';
export * from './Close';
