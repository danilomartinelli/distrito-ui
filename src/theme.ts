import { DefaultTheme } from 'styled-components';

export const lightTheme: DefaultTheme = {
  name: 'Light Theme',
  font: {
    weight: {
      regular: '400',
      medium: '500',
      bold: '700',
    },
  },
  colors: {
    brand: {
      n700: '#a40047',
      n600: '#cc0057',
      n500: '#e50063',
      n200: '#ff99c5',
      n100: '#ffebf3',
    },
    neutral: {
      white: '#ffffff',
      n100: '#f1f3f5',
      n200: '#e9ecef',
      n300: '#dee2e6',
      n400: '#ced4da',
      n500: '#adb5bd',
      n800: '#343a40',
    },
  },
};

export const themeArgs = {
  theme: {
    description: 'Distrito themes',
    defaultValue: lightTheme.name,
    options: [lightTheme.name],
    control: { type: 'select' },
    mapping: {
      [lightTheme.name]: lightTheme,
    },
  },
};
