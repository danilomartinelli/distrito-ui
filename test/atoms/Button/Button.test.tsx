import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { lightTheme, Button } from '../../../src';

test('Renders correctly', () => {
  const component = renderer.create(
    <ThemeProvider theme={lightTheme}>
      <Button label="Click here!" variant="raised" />
    </ThemeProvider>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
