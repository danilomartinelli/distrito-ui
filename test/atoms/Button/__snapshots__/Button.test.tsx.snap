// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<button
  className="sc-pNWdM kYPiiX"
  disabled={false}
  type="button"
>
  <span
    className="sc-fujyAs ioZBJQ"
  >
    Click here!
  </span>
</button>
`;
